controllers/ExportController.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+namespace humhub\modules\legal\controllers;
+
+use Yii;
+use humhub\modules\user\components\BaseAccountController;
+use humhub\modules\post\models\Post;
+use yii\web\Response;
+
+/**
+ * ExportController handles exporting user data as JSON.
+ *
+ */
+class ExportController extends BaseAccountController
+{
+    /**
+     * Displays the exported user data as JSON.
+     *
+     * @return string The rendered view.
+     */
+    public function actionIndex()
+    {
+        $jsonData = $this->getUserDataAsJson();
+
+        // Render the view with the JSON data
+        return $this->renderAjax('index', ['jsonData' => $jsonData]);
+    }
+
+    /**
+     * Downloads the exported user data as a JSON file.
+     *
+     * @return Response The file response.
+     */
+    public function actionDownload()
```

Comment:
Optional idea for additional security, should we hash the generated JSON file?

Reply:
Hmm, not sure if this gives a significant security benefit.

Reply:
> Hmm, not sure if this gives a significant security benefit.

Was just an idea to do this on transmission, but I guess you are right.